File: backend/schemas/requests.py
from typing import Literal
from pydantic import BaseModel, Field
from models.domain import CompanyProfile, BillingProfile, ProjectContext

class UserCreate(BaseModel):
    full_name: str
    email: str
    password: str = Field(min_length=8)
    company: CompanyProfile
    billing: BillingProfile | None = None

class UserUpdate(BaseModel):
    full_name: str | None = None
    email: str | None = None
    company: CompanyProfile | None = None
    billing: BillingProfile | None = None

class ProjectCreate(BaseModel):
    name: str
    context: ProjectContext

class ProjectUpdate(BaseModel):
    name: str | None = None
    context: ProjectContext | None = None

class AgentCreate(BaseModel):
    name: str
    age_range: str | None = None
    segment: str
    motivations: list[str] = Field(default_factory=list)
    objections: list[str] = Field(default_factory=list)

class SimulationCreate(BaseModel):
    scenario_name: str
    questions: list[str] = Field(default_factory=list)
    overrides: dict[str, str | int | float | bool] = Field(default_factory=dict)
    provider: Literal["openai", "gemini", "claude", "mock"] = "mock"

File: backend-go/schemas/requests.go
package schemas

import (
	"backend-go/models"
)

type UserCreate struct {
	FullName string                `json:"full_name"`
	Email    string                `json:"email"`
	Password string                `json:"password"`
	Company  models.CompanyProfile `json:"company"`
}

type LoginRequest struct {
	Email    string `json:"email"`
	Password string `json:"password"`
}

type UserUpdate struct {
	FullName *string                `json:"full_name,omitempty"`
	Email    *string                `json:"email,omitempty"`
	Company  *models.CompanyProfile `json:"company,omitempty"`
}

type ProjectCreate struct {
	Name    string                `json:"name"`
	Context models.ProjectContext `json:"context"`
}

type ProjectUpdate struct {
	Name    *string                `json:"name,omitempty"`
	Context *models.ProjectContext `json:"context,omitempty"`
}

type TargetModelCreate struct {
	Name               string   `json:"name"`
	Age                *int     `json:"age,omitempty"`
	Occupation         *string  `json:"occupation,omitempty"`
	SocioeconomicLevel *string  `json:"socioeconomic_level,omitempty"`
	Personality        []string `json:"personality"`
}

type TargetModelUpdate struct {
	Name               *string   `json:"name,omitempty"`
	Age                *int      `json:"age,omitempty"`
	Occupation         *string   `json:"occupation,omitempty"`
	SocioeconomicLevel *string   `json:"socioeconomic_level,omitempty"`
	Personality        *[]string `json:"personality,omitempty"`
}

type AgentCreate struct {
	Name        string   `json:"name"`
	Gender      *string  `json:"gender,omitempty"`
	Segment     string   `json:"segment"`
	Motivations []string `json:"motivations"`
	Objections  []string `json:"objections"`
}

type AgentUpdate struct {
	ModelID     *string   `json:"model_id,omitempty"`
	Name        *string   `json:"name,omitempty"`
	Gender      *string   `json:"gender,omitempty"`
	Segment     *string   `json:"segment,omitempty"`
	Motivations *[]string `json:"motivations,omitempty"`
	Objections  *[]string `json:"objections,omitempty"`
}

type Provider string

const (
	ProviderOpenAI Provider = "openai"
	ProviderGemini Provider = "gemini"
	ProviderClaude Provider = "claude"
	ProviderMock   Provider = "mock"
)

type SimulationCreate struct {
	ScenarioName string   `json:"scenario_name"`
	Questions    []string `json:"questions"`
	Provider     Provider `json:"provider"`
}
